bitcoin/src/blockdata/script.rs

```diff
@@ -139,13 +139,19 @@ pub struct Script([u8]);
 ///
 /// [deref coercions]: https://doc.rust-lang.org/std/ops/trait.Deref.html#more-on-deref-coercion
 #[derive(Default, Debug, Clone, PartialOrd, Ord, PartialEq, Eq, Hash)]
-pub struct ScriptBuf(Vec<u8>);
+pub struct ScriptBuf<T = Generic>(PhantomData<T>, Vec<u8>);
```

Comment:
Why did you put `PhantomData` first? I always saw it last in all codebases and it'd also avoid renumbering some field accesses.

Reply:
Ha, this was a tedious lesson to learn indeed - the last item in a tuple struct can be unknown size but no other so if `PhantomData` is last we require explicit lifetimes everywhere on the vec - which I put before working out to put phantom data first. (I actually did it on a stripped down version of the script module on `Script`.)

Reply:
My bad, I should have commented on that in the PR description.

Reply:
Ah, true, it can't be last in `Script` but can be in `ScriptBuf`. Would've saved a bunch of diff lines but if you want to keep it for consistency that's OK.

Reply:
Ah you were too kind, I realised I misspoke yesterday while doing my morning routine this morning and was fully expecting to come to a message saying "Tobin you wombat, `Vec` _is_ `Sized`"  :)

Reply:
I'll wait till we resolve `Script` then see whats best.